modules/connectivity/ParticipantConnectionStatus.js

```diff
@@ -624,40 +648,56 @@ export default class ParticipantConnectionStatusHandler {
                 isInLastN} currentStatus => newStatus: ${
                 participant.getConnectionStatus()} => ${newState}`);
 
-        const nowMs = new Date().getTime();
-        const internalState = {
-            'instantMs': nowMs,
-            'isVideoMuted': isVideoMuted,
-            'isConnActiveByJvb': isConnActiveByJvb,
-            'isVideoTrackFrozen': isVideoTrackFrozen,
-            'inP2PMode': inP2PMode,
-            'inInLastN': isInLastN
-        };
-
-        if (!this.peerConnStatusMap[id] || this.peerConnStatusMap[id].state !== newState) {
-            // The peer connection status has changed. Compute the duration of the current
-            // connection status and send it as an analytics event.
-            if (this.peerConnStatusMap[id]) {
-                this.peerConnStatusMap[id].value = nowMs - this.peerConnStatusMap[id].startedMs;
-                Statistics.sendAnalytics(
-                    createParticipantConnectionStatusEvent(this.peerConnStatusMap[id]));
-            }
+        const oldConnectionStatus = this.connectionStatusMap[id] || {};
 
-            // And start a new status for the participant.
-            this.peerConnStatusMap[id] = {
-                'internalStates': [ internalState ],
-                'state': newState,
-                'startedMs': nowMs
-            };
-        } else {
-            // The connection status hasn't changed, but there was an internal state change.
-            // Register the internal state.
-            this.peerConnStatusMap[id].internalStates.push(internalState);
-        }
+        // Send an analytics event (guard on either the p2p flag or the connection status has changed
+        // since the last time this code block run).
+        if (!('p2p' in oldConnectionStatus)
+            || !('connectionStatus' in oldConnectionStatus)
+            || oldConnectionStatus.p2p !== inP2PMode
+            || oldConnectionStatus.connectionStatus !== newState) {
+
+            const nowMs = Date.now();
+
+            this.maybeSendParticipantConnectionStatusEvent(id, nowMs);
 
+            this.connectionStatusMap[id] = Object.assign(oldConnectionStatus, {
```

Comment:
you can use:
```
this.connectionStatusMap[id] = {
    ...oldConnectionStatus,
     connectionStatus: newState,
     p2p: inP2PMode,
     startedMs: nowMs
};
```

Seems a little bit more readable to me :) WDYT?

Reply:
sure, just to make sure i understand, you mean not use `Object.assign` but copy the fields one by one instead?

Reply:
not one by one but using the spread operator(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). you can just copy the code from above :)

But this is optional, so if you prefer the `Object.assign` please keep it!

Reply:
i feel like an old-schooler but TIL :)